pandas/core/series.py

```diff
@@ -4153,4 +4153,8 @@ def _try_cast(arr, take_fast_path):
                 data = np.array(data, dtype=dtype, copy=False)
             subarr = np.array(data, dtype=object, copy=copy)
 
+            # GH 20377
```

Comment:
huh? this should *not* be necessary (not to mention non-performant)

Reply:
Well, this fixes an error I got in this test https://github.com/pandas-dev/pandas/blob/master/pandas/tests/frame/test_dtypes.py#L533

This line https://github.com/nikoskaragiannakis/pandas/blob/7341cd17e11461728969afa159250e882b32dee0/pandas/core/series.py#L4154 does the damage by turning a `np.nan` to `'nan'`. But this comes directly from numpy, so I cannot change it.

Am I missing something?

Reply:
how's that an error? that test is correct.

Reply:
I think that with the changes in the .pyx files, `np.nan` stays a float, even if you try to cast it to `str`.
For example
```
pd.DataFrame([np.NaN]).astype(str)
```
leaves `np.NaN` a float, while before the changes you used to get a `'nan'`.

So I have to ask: do we want calls like the one above to turn `np.NaN` into `nan` or not?

Reply:
@jreback ^